src/auxiliary/p4est.jl

```diff
@@ -38,24 +47,23 @@ function unsafe_wrap_sc(::Type{T}, sc_array_obj::sc_array) where T
   return unsafe_wrap(Array, Ptr{T}(array), elem_count)
 end
 
+function unsafe_wrap_sc(::Type{T}, sc_array_pw::PointerWrapper{sc_array}) where T
+  elem_count = sc_array_pw.elem_count[]
+  array = sc_array_pw.array
 
-# Load the ith element (1-indexed) of an sc array of type T
-function unsafe_load_sc(::Type{T}, sc_array::Ptr{sc_array}, i=1) where T
-  sc_array_obj = unsafe_load(sc_array)
-  return unsafe_load_sc(T, sc_array_obj, i)
+  return unsafe_wrap(Array, Ptr{T}(pointer(array)), elem_count)
 end
 
-function unsafe_load_sc(::Type{T}, sc_array_obj::sc_array, i=1) where T
-  element_size = sc_array_obj.elem_size
-  @assert element_size == sizeof(T)
 
-  return unsafe_load(Ptr{T}(sc_array_obj.array), i)
+# Load the ith element (1-indexed) of an sc array of type T as PointerWrapper
+function unsafe_load_sc(::Type{T}, sc_array::PointerWrapper{sc_array}, i::Integer=1) where T
+  return PointerWrapper(T, pointer(sc_array.array) + (i - 1) * sizeof(T))
 end
```

Comment:
I'm not sure this is the best name for this function. It behaves differently than `unsafe_load` would, doesn't it? In particular, one needs an additional `[]` when using this, correct?

Reply:
Yes, you're right. What about `load_pointerwrapper_sc`?

Reply:
Sounds good to me :+1:

Reply:
Done in [d141a8b](https://github.com/trixi-framework/Trixi.jl/pull/1434/commits/d141a8b33f59469886d2759c656b095e310a4c68).